### ISSUE TITLE
The tick: two phases, unit delay, external writes

### ISSUE THREAD

**Issue by NGL321** [NGL321/patchworks#86]
## What to build

The tick. This is the first tracer bullet through the whole system: the world writes boundary cells, the graph runs, a torque leaves by the actuator's motor edge, and the world moves. The body is untrained and the maps are at their initial values, so the arm will flail — that is expected and is not what this ticket is judged on.

**Every tick is exactly two phases, in this order.**

1. **Inference phase.** Every cell, simultaneously and independently, runs its forward path — `encode` fuses its persisted chart with the node stalk the previous message-passing phase left behind, `step` advances it one tick, `decode` reads off the predicted node stalk. No cell reads another cell's state here.
2. **Message-passing phase.** Every cell restricts its own predicted node stalk onto each incident edge and runs **one** local reconciliation step against the belief its neighbour restricted onto that same edge. The result edits the **node stalk only, never the chart**.

**External writes are an ordering, not a third phase.** Whatever is outside the sheaf writes its boundary cells *after* the message-passing phase, as the last word in a tick. Nothing computes during it. Two things fall out without machinery: a drive's standing assertion actually stands, because reconciliation can only ever have its disagreement reduced by the *cell* moving; and the motor pathway is untouched, because the actuator's three commanded components are written by nobody outside, so reconciliation fills them and the world reads them. A blanket "boundary stalks are not reconciled" rule would have severed that; the ordering rule needs no exemption.

**Delay: an edge costs exactly one tick.** At tick `t` a cell combines its own restriction, computed this tick, with its neighbour's restriction as that neighbour broadcast it at `t − 1`. What a cell reconciles against is always one tick stale.

**One step, not a solve.** Exactly one simultaneous local descent step per cell per tick — not a tunable round count and not iterate-to-convergence. All cells read the same prior round's incoming values and update at once, Jacobi-style, so there is no visiting order to define. Per-tick cost is bounded by construction: no convergence check, no early stopping.

**Reconciliation gain**, per cell, normalised by total incident mask width:

```
gain_v  =  γ / max( Σ_{e∋v} m_e , ρ² · deg(v) )     with a single global γ ≤ 1
```

Written as the max so a later change to `ρ` cannot silently loosen the bound. This **equalises** the effective step across the graph and removes a degree artifact — it is **not a timescale knob and must not become one**. A gain deliberately graded by depth would be the per-cell clock divisor ADR-0005 rejected, wearing a different name.

**The tick carries no tape.** The whole tick runs under `torch.no_grad()`. This is not an optimisation: the tick is a rollout, not a training pass, and every quantity it produces is *data* the cell then learns from.

## Acceptance criteria

- [ ] Two phases run in order every tick, with external writes landing after message passing and winning
- [ ] Edge buffers carry unit delay: a cell reconciles against its neighbour's `t − 1` broadcast
- [ ] Exactly one Jacobi-style descent step per cell per tick, with no convergence check anywhere in the loop
- [ ] `gain_v` is computed as the specified max, with a single global `γ`
- [ ] The whole tick runs under `no_grad`, and a cheap always-on assertion confirms nothing leaving the tick carries a `grad_fn`
- [ ] An untrained agent drives the sandbox for a sustained run without divergence or crash, emitting torques through the actuator boundary cell
- [ ] Per-tick wall time is benchmarked against the ~3.18 ms `env.step()` figure and reported

## Spec sources

- `docs/spec/02-tick-semantics.md` — the whole file
- `docs/spec/09-the-build-stack.md` — *The locality guard*, part 1
- `docs/spec/04-action-and-the-boundary.md` — *The boundary*
- `docs/adr/0002-message-passing-is-one-step-not-a-solve.md`, `docs/adr/0003-action-is-prediction-the-world-clears.md`

## Blocked by

- #BLOCK04
- #BLOCK05
- #BLOCK06

**Comment by NGL321** [comment]
Merged as `3fd6bd8` on `action`; ticket commit `95d96d3` on branch `worktree-agent-a7a0555a6de60abd1`, branched from `10cb8e6`. Clean merge, no conflicts. Re-tested on `action` after the merge: **246 pass** (168 before, 78 new).

Built: `src/patchworks/restriction.py` (one masked, gauge-fixed map per edge endpoint, one padded `[1364, 8, 48]` tensor so restriction is a single `bmm`), `src/patchworks/tick.py` (`StalkLayout`, `Sheaf`, `reconciliation_gain`, `assert_no_tape`; ragged stalks in one flat buffer, no loop over cells or edges), `src/patchworks/agent.py` (the ordering, the rim translation, `run()`), plus `benchmarks/agent_tick.py` and a measured row in `09-the-build-stack.md`.

Both of #77's load-bearing items were built as specified. The tape assertion runs at the end of each phase, the end of the tick, and after the external write; `torch.no_grad()` is a context manager rather than a decorator so `mock.patch` can still reach it, and a mutation test confirms the guard is load-bearing rather than decorative — unguarded, `message_passing_phase` alone taints `stalks` with `SubBackward0` and `inference_phase` with `IndexPutBackward0`. For `gain_v`, two tests: multiplying the gain back out returns exactly `γ` for every cell, and every structural class spanning two levels of the taper has one gain value — nowhere for a per-cell knob to hide.

**Open handoff, carried to #85: `γ`'s value is provisional.** `DEFAULT_GAMMA = 1.0`, the spec's cap. `02-tick-semantics.md` also requires `γ × floor` to stay below the tightest cell's fold margin, and assigns that check to the construction-time bias-sampling rig — #85. Recorded on #85 so the number comes back and sets this default.

Two observations from the agent, neither a defect. **The run settles**: over 2000 ticks the arm pins against its joint limits and reaches a fixed point, identical torques to seven decimals, disagreement holding at ~2190 and never cleared. That is cross-tick settling under a frozen contractive body with nothing learning, which `02-tick-semantics.md` *Known exposure* says is to be observed during the build rather than designed for. #88/#89 are expected to break it. **The benchmark needs an idle laptop**: the recorded 0.92 ms against a 3.37 ms `env.step()` came from a quiet window; under contention the tick spreads over six torch threads and `env.step()` does not, so only the total and the two contention-surviving ratios went into the spec table.

**Comment by NGL321** [comment]
Correction to the handoff recorded above: **#85 cannot pin `DEFAULT_GAMMA`,** so the provisional `1.0` stands until a human rules.

#85 ran its fold-margin sweep (branch `worktree-agent-a13d46ce562b23297`, HEAD `5de1ed5`, held unmerged). The check caps the *product* `γ × floor`, not `γ` — the apex binds at `γ × floor ≤ 0.260` — and the missing input is a **disagreement floor**, which is a runtime magnitude that nothing trains and that ADR-0007 has nothing representing. `γ = 1.0` remains admissible exactly while the floor stays below 0.260; at a unit floor the constant would be 0.260.

Details and the competing reading (which moves the constant by 20×) are on #85.
